src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObjectFactory.Windows.cs

```diff
@@ -8,29 +8,43 @@ namespace System.Data.SqlClient
 {
     internal sealed class TdsParserStateObjectFactory
     {
-        public static bool useManagedSni = false;
+
+        private const string UseLegacyNetworkingOnWindows = "System.Data.SqlClient.UseLegacyNetworkingOnWindows";
 
         public static readonly TdsParserStateObjectFactory Singleton = new TdsParserStateObjectFactory();
 
+        static TdsParserStateObjectFactory()
+        {
+            bool shouldUseLegacyNetorking;
+            // If The appcontext switch is set then Use Managed SNI based on the value. Otherwise Managed SNI should always be used.
+            UseManagedSNI = AppContext.TryGetSwitch(UseLegacyNetworkingOnWindows, out shouldUseLegacyNetorking) ? !shouldUseLegacyNetorking : true;
+        }
```

Comment:
Nit: you could avoid some perf overheads by removing the explicit static cctor, removing the private setter, and changing this all to something like:
```C#
public static bool UseManagedSNI { get; } =
    AppContext.TryGetSwitch(UseLegacyNetworkingOnWindows, out bool shouldUseLegacyNetworking) ? !shouldUseLegacyNetworking : true;
```